File: Frontend/MptRoomFrontend/src/app/models/DTO/lesson-supersede-request.dto.ts
import { SupersedeRequestStatus } from '../enums/supersede-request-status.enum';
import { SupersedeRequestType } from '../enums/supersede-type-request.enum';

export interface LessonSupersedeRequestDTO {
  supersedeRequestId?: number;
  teacherId: number;
  groupId: number;
  dateToSupersede: string;
  lessonSlotId: number;
  subjectId: number;
  requestTime: string;
  affectedLessonId?: number;
  supersedeRequestStatus: SupersedeRequestStatus;
  supersedeRequestType: SupersedeRequestType;
}

File: Backend/MptRoomAPI/MptRoomAPI/DTO/LessonSupersedeRequestDTO.cs
using MptRoomAPI.Models.Enums;

namespace MptRoomAPI.DTO
{
    public class LessonSupersedeRequestDTO
    {
        public int? SupersedeRequestId { get; set; }
        public int TeacherId { get; set; }
        public int GroupId { get; set; }
        public DateOnly DateToSupersede { get; set; }
        public int LessonSlotId { get; set; }
        public int SubjectId { get; set; }
        public DateTime RequestTime { get; set; }
        public int? AffectedLessonId { get; set; } 
        public SupersedeRequestStatusEnum SupersedeRequestStatus { get; set; }
        public SupersedeRequestTypeEnum SupersedeRequestType { get; set; }

    }
}
